### PR TITLE
lantiq-xrx200: add support for TP-Link TD-W8970, TD-W8980 and TD-W9980

### PR DESCRIPTION
**Pull request by kpanic23**
* [x]  must be flashable from vendor firmware
  
  * [x]  webinterface (if old enough)
  * [o]  tftp
  * [x]  other: webinterface shell injection (if newer stock firmware)

* [x]  must support upgrade mechanism
  
  * [x]  must have working sysupgrade
    
    * [x]  must keep/forget configuration (if applicable)
      _think `sysupgrade [-n]` or `firstboot`_
  * [?] must have working autoupdate
    _usually means: gluon profile name must match image name_

* [x]  wps button must return device into config mode

* [x]  primary mac should match address on device label (or packaging) (https://gluon.readthedocs.io/en/latest/dev/hardware.html#notes)

* wired network
  
  * [x]  should support all network ports on the device
  * [x]  must have correct port assignment (WAN/LAN)

* wifi (if applicable)
  
  * [x/o]  association with AP must be possible on all radios
  * [x/o]  association with 802.11s mesh must be working on all radios
  * [x/o]  ap/mesh mode must work in parallel on all radios

* led mapping
  
  * power/sys led (_critical, because led definitions are setup on firstboot only_)
    
    * [x]  lit while the device is on
    * [x]  should display config mode blink sequence
      (https://gluon.readthedocs.io/en/latest/features/configmode.html)
  * radio leds
    
    * [x]  should map to their respective radio
    * [x]  should show activity
  * switchport leds
    
    * [x]  should map to their respective port (or switch, if only one led present)
    * [x]  should show link state and activity

* outdoor devices only
  
  * [n/a]  added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`

### CODE REVIEW

File: targets/lantiq-xrx200 [hunk 1]
```diff
@@ -1,3 +1,12 @@
 device('avm-fritz-box-7412', 'avm_fritz7412', {
         factory = false,
 })
+
+-- TP-Link
+
+device('tp-link-td-w8970', 'tplink_tdw8970')
```
Comment (blocktrron) [inline targets/lantiq-xrx200]: This currently still needs a BROKEN flag, as we have not yet made a decision on how to handle hybrid LAN/WAN ports.

Comment (kpanic23) [reply targets/lantiq-xrx200]: Done :)

**Comment by kpanic23** [comment]
Is there anything still missing?

**Comment by rotanid** [comment]
> Is there anything still missing?

maybe similar discussion about the ports as in #1882 ?

File: targets/lantiq-xrx200 [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+-- TP-Link
+
+device('tp-link-td-w8970', 'tplink_tdw8970', {
+	broken = true, -- We have not yet made a decision on how to handle hybrid LAN/WAN ports.
+})
+
+device('tp-link-td-w8980', 'tplink_tdw8980', {
+	aliases = {'tp-link-td-w9980'},
+	broken = true, -- 5GHz unsupported
+})
```
Comment (rotanid) [inline targets/lantiq-xrx200]: GitHub marks a missing newline at end-of-file, please check if there was one before and in the other target files and if yes, please add the newline (newline != empty line)

**Comment by blocktrron** [comment]
We've talked about the hybrid-port scenario and came to the conclusion, which is outlined in #1886 . Can you confirm the tp-link-td-w8970 matches the behavior described in the pull-request.

**Comment by rubo77** [comment]
shouldnt we change the commit message to TD-W8970 instead of TL-W8970?

**Comment by blocktrron** [comment]
On Reviewday, it was concluded, that the device (`td-w8970`) qualifies as non-broken.

BUT could you please add a comment about the LAN-port strangenes in the target file?

**Comment by mweinelt** [comment]
> BUT could you please add a comment about the LAN-port strangenes in the target file?

The reason should be prefixed with "CAVEAT:", similar to the reasons for brokenness.

**Comment by kpanic23** [comment]
> > BUT could you please add a comment about the LAN-port strangenes in the target file?
> 
> The reason should be prefixed with "CAVEAT:", similar to the reasons for brokenness.

Done!

**Comment by rotanid** [comment]
@kpanic23 please add the non-broken device to the supported devices list

File: targets/lantiq-xrx200 [hunk 3]
```diff
@@ -10,3 +10,17 @@ device('avm-fritz-box-7362-sl', 'avm_fritz7362sl', {
 device('avm-fritz-box-7412', 'avm_fritz7412', {
         factory = false,
 })
+
+-- TP-Link
+
+  -- CAVEAT: These devices don't have a dedicated WAN port.
+  --         All ethernet ports on the device are bridged and
+  --         used as WAN ports. Clients connected to these
+  --         ports will be connected to your private network.
```
Comment (mweinelt) [inline targets/lantiq-xrx200]: Meh, I think this isn't user-visible enough and we should rather make that a note in the supported device list. Sorry this is coming so late.

Comment (mweinelt) [reply targets/lantiq-xrx200]: We already have the `lan_as_wan` note, so I'm going to add that.

Comment (rotanid) [reply targets/lantiq-xrx200]: thanks

**Comment by rotanid** [comment]
@kpanic23 please test-build your PRs... there was another problem, as these devices don't have factory image, @mweinelt fixed this in 694b6e493dd507f0b8a6516ae58c0be3f89ed8aa

**Pull request merged**